src/analysis/vector/geometry_checker/qgsgeometryoverlapcheck.cpp

```diff
@@ -153,7 +153,9 @@ void QgsGeometryOverlapCheck::fixError( const QMap<QString, QgsFeaturePool *> &f
   }
   else if ( method == Subtract )
   {
-    std::unique_ptr< QgsAbstractGeometry > diff1( geomEngineA->difference( interPart, &errMsg ) );
+    std::unique_ptr< QgsGeometryEngine > geomEngineDiffA = QgsGeometryCheckerUtils::createGeomEngine( geometryA.constGet(), 0 );
+    geomEngineDiffA->prepareGeometry();
```

Comment:
There's no benefit to this -- the GEOS difference method doesn't take advantage of prepared geometries, so it's just extra work to prepare them.